extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/TokensHelper.java

```diff
@@ -0,0 +1,76 @@
+package io.quarkus.oidc.client.runtime;
+
+import java.util.concurrent.locks.ReentrantLock;
+
+import io.quarkus.oidc.client.OidcClient;
+import io.quarkus.oidc.client.Tokens;
+import io.smallrye.mutiny.Uni;
+
+public class TokensHelper {
+
+    private volatile Tokens tokens;
+    private ReentrantLock tokenAcquisitionLock = new ReentrantLock();
```

Comment:
We can't really do this via locks when you are dealing with reactive code. getTokens() can be called by the same thread for reactive code, but for different requests. The logic below also has a few issues.

Reply:
@stuartwdouglas, thanks a lot; I have to admit this possibility did not occur to me :-) - I was primarily focused on running this code from the client filter which at this moment of time is not reactive ready yet as if I understand it correctly - Steph said he was only planning to support the reactive filters in 'regular' RestEasy - but you are right it has to be done right - as I'd like to have these filters running eventually in the reactive mode :-)
Also, thanks for getting this code right, let me learn what has changed, cheers

Reply:
@stuartwdouglas - compare and swap :-) - I'm getting inspired here and I've just pulled Java Concurrentcy in Practice 2nd edition from the bookshelf though I suspect it may've gone a bit outdated :-)

Reply:
@stuartwdouglas - actually - I'm finally getting how to make sure `Uni` can be kept as a state property, it must be `deferred`, I've been thinking of keeping it in `quarkus-oidc` somewhere as a state value but could not get how to do it :-)